channels/serializers.py

```diff
@@ -149,6 +149,20 @@ def _get_user(self, instance):
         else:
             return User.objects.filter(username=instance.author.name).select_related('profile').first()
 
+    def _get_profile(self, instance):
+        """ Return a user profile if it exists, else None
+        Args:
+            instance (praw.models.Submission):
+                The post to look up the user for
+
+        Returns:
+            profiles.models.Profile: the user profile if it exists
+        """
+        user = self._get_user(instance)
+        if user and hasattr(user, 'profile'):
```

Comment:
I don't think it's possible for a user not to have a `profile` property at all. If there's no associated profile in the db, this is just `None`, right?

Reply:
OK. In that case, I think it would definitely be considered exceptional that a user does not have an associated profile. Given that, it's a bit more pythonic to implement this as a try/catch and return `None` when `RelatedObjectDoesNotExist` is raised